### PR TITLE
fix: Fix issue 22837 about unknown error during ledger pair

### LINKED ISSUE #22837: [Bug]: Unable to pair to Ledger a second time after removing accounts
### Describe the bug

chrome 117.0.5938.92 
mac Sonoma 14.2.1 
Ledger Nano X running firmware 2.1.0 
Ethereum app 1.10.1

If I pair the Ledger device, remove the accounts, and try to pair again I am stuck with an `Unknown error` and unable to add any further Ledger accounts.  

The only way I have found so far to get out of this situation is to [forget all HID devices](chrome://settings/content/hidDevices) and reload the extension.

This was observed in 11.10 RC testing, but confirmed to exist in 11.9.0. 

### Expected behavior

Should be able to add and remove Ledger accounts repeatedly without having to forget devices and reload extension.

### Screenshots/Recordings

https://recordit.co/P9YMp1mOcN

### Steps to reproduce

1. Setup wallet
2. Connect, unlock and open the Ethereum app on Ledger
3. Use the add hardware flow to add some Ledger accounts
4. Remove the Ledger accounts from full screen mode
5. Use add hardware flow to try and add any Ledger accounts
6. Note Unknown Error in extension

### Error messages or log output

```shell
LEDGER:::CREATE APP ERROR 
TransportError {name: 'TransportError', message: "U2F browser support is needed for Ledger. Please u…ion. Also make sure you're on an HTTPS connection", id: 'U2FNotSupported', stack: 'Error\n    at new TransportError (https://metamask.…b.io/eth-ledger-bridge-keyring/bundle.js:22460:22'}
id
: 
"U2FNotSupported"
message
: 
"U2F browser support is needed for Ledger. Please use Chrome, Opera or Firefox with a U2F extension. Also make sure you're on an HTTPS connection"
name
: 
"TransportError"
stack
: 
"Error\n    at new TransportError (https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:18069:18)\n    at https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:22460:22"
[[Prototype]]
: 
Error at new TransportError (https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:18069:18) at https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:22460:22
stack
: 
"Error\n    at new TransportError (https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:18069:18)\n    at https://metamask.github.io/eth-ledger-bridge-keyring/bundle.js:22460:22"
[[Prototype]]
: 
Object
constructor
: 
ƒ Error()
message
: 
""
name
: 
"Error"
toString
: 
ƒ toString()
[[Prototype]]
: 
Object
constructor
: 
ƒ Object()
hasOwnProperty
: 
ƒ hasOwnProperty()
isPrototypeOf
: 
ƒ isPrototypeOf()
propertyIsEnumerable
: 
ƒ propertyIsEnumerable()
toLocaleString
: 
ƒ toLocaleString()
toString
: 
ƒ toString()
valueOf
: 
ƒ valueOf()
__defineGetter__
: 
ƒ __defineGetter__()
__defineSetter__
: 
ƒ __defineSetter__()
__lookupGetter__
: 
ƒ __lookupGetter__()
__lookupSetter__
: 
ƒ __lookupSetter__()
__proto__
: 
(...)
get __proto__
: 
ƒ __proto__()
set __proto__
: 
ƒ __proto__()
```

### Version

11.9.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

Ledger

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**

Issue #22837 descrip that ledger sometimes will display `Unknown Error` during pairing. we have tried to replicate the issue and discover that it happen during ledger is lock and not open Eth app. this PR will replace the `Unknown Error` with more meaningful error message to guide user solve the issue.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25462?quickstart=1)

## **Related issues**

Fixes: #22837 

## **Manual testing steps**

1. Setup wallet
2. Connect, unlock and open the Ethereum app on Ledger
3. Use the add hardware flow to add some Ledger accounts
4. Remove the Ledger accounts from full screen mode
5. Use add hardware flow to try and add any Ledger accounts
6. Select the `paired` with that ledger when it is available
7. ledger should be locked status or unlock ledger but not opening eth app.
8. you will see new error message `Unlock your Ledger device and open the ETH app`
9. After you unlock your ledger and open ETH app. and click paired again
10. you should be able to select accounts from next screen and import those accounts from ledger.

### **Before**
https://recordit.co/kdCDL4laWo

### **After**
https://github.com/MetaMask/metamask-extension/assets/7315988/f11a1867-ad3f-4796-8661-6d3469bed682

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .yarn/patches/@metamask-eth-ledger-bridge-keyring-npm-2.0.1-7a5d815b2d.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+             });
+         }
+         catch (error) {
+-            throw error instanceof Error ? error : new Error('Unknown error');
++         
++            /**
++             * For Fixing issue 22837, when ledger is locked and didnt open the ethereum app in ledger,
++             * The extension will always show `unknown error`, below change will transform the error to something meaningful.
++             */
++            throw error instanceof Error ? error : new Error('Please Unlock your ledger and open ETH app');
```
Comment (vivek-consensys) [inline .yarn/patches/@metamask-eth-ledger-bridge-keyring-npm-2.0.1-7a5d815b2d.patch:15]: @dawnseeker8 can we use 'Unlock your Ledger device and open the ETH app' ?

Comment (dawnseeker8) [reply .yarn/patches/@metamask-eth-ledger-bridge-keyring-npm-2.0.1-7a5d815b2d.patch:15]: ok, i will do a change on the message.

**Review by vivek-consensys — APPROVED** [review]

**Comment by vivek-consensys** [comment]
Built locally and tested, working as expected.

**Review by chloeYue — APPROVED** [review]
LGTM !

**Pull request merged**